### ISSUE TITLE
[do first] Spike/Do: token retrieval for Cypress [login utility method]

### ISSUE THREAD

**Issue by znikola** [SAP/cloud-commerce-spartacus-storefront#1216]
- we want to create only 1 user per test suite, and then retrieve the token for usage later in the test suite
- this user creation should be a utility method
- this will be used by all tests/testsuites
- should have a mechanism to create a new one if you don't want to use the one you got or if you need more than 1
- should have a mechanism for anonymous user
- user should be random email and new every time

can use what marcin already has in big happy path.
all tests that don't require login can be started.
we'll identify the tests that require login and wait for this to be done.

---- previous description ----
Implement token retrieval in `before` method and expose it via `getToken()` method, so that other tests can just use the token and directly navigate (deep link) to desired page without having to go through all the steps every time (register -> login -> etc.).

This should be done before doing other cypress tickets.

Also, see master ticket: #1102 

all epic/cypressmigration tickets should be merged into epic branch "epic/cypressmigration-fr2".
if this epic branch doesn't exist, you're probably the first, so create it.

**Comment by Xymmer** [comment]
Marcin will update this ticket with a code snippet that would take care of this one.

**Comment by Xymmer** [comment]
for now, we are creating a new user for each test suite/run

**Comment by marlass** [comment]
Snippet from knowledge transfer session:
``` ts
before(() => {
    cy.window().then(win => win.sessionStorage.clear());
    cy.server();
    cy.request({
      method: 'POST',
      url:
        'https://storefront.c39j2-walkersde1-d4-public.model-t.cc.commerce.ondemand.com/authorizationserver/oauth/token',
      body: {
        client_id: 'mobile_android',
        client_secret: 'secret',
        grant_type: 'password',
        username: 'user14359918131@ydev.hybris.com',
        password: 'Password123.'
      },
      form: true
    }).then(res => {
      const a = {
        userToken: {
          token: { ...res.body, userId: 'user14359918131@ydev.hybris.com' }
        },
        clientToken: {
          loading: false,
          error: false,
          success: false
        }
      };
      cy.window().then(win => {
        win.sessionStorage.setItem('auth', JSON.stringify(a));
        cy.visit('/');
      });
    });
  });
```

It was only done for demo purposes. It only presents idea. Code should be refactored for final use.
Main problem I find with this code is static user. We also need code to register user if it doesn't exist. 
And user should be the same for each spec (would require using Cypress environment variable and creating it in `support/index.ts` file) or we stick with having separate user for each file (also good approach).
I am good with both approaches as long as whole login/register flow is headless (no UI interaction).
